controllers/utils/storageclass.go

```diff
@@ -0,0 +1,148 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	storagev1 "k8s.io/api/storage/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type CephType string
+
+const (
+	RBDType                            CephType = "rbd"
+	CephFSType                         CephType = "cephfs"
+	RBDProvisionerTemplate                      = "%s.rbd.csi.ceph.com"
+	CephFSProvisionerTemplate                   = "%s.cephfs.csi.ceph.com"
+	defaultCephRBDStorageClassTemplate          = "%s-ceph-rbd"
+	defaultCephFSStorageClassTemplate           = "%s-cephfs"
+	DefaultSubvolumeGroup                       = "csi"
+	DefaultRadosNamespace                       = ""
+)
+
+func GetStorageClass(ctx context.Context, c client.Client, storageClassName string) (*storagev1.StorageClass, error) {
+	storageClass := &storagev1.StorageClass{}
+
+	err := c.Get(ctx, client.ObjectKey{
+		Name: storageClassName,
+	}, storageClass)
+
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, err
+		}
+		return nil, fmt.Errorf("failed to fetch StorageClass %s : %v", storageClassName, err)
+	}
+
+	return storageClass, nil
+}
+
+func GetDefaultStorageClasses(ctx context.Context, c client.Client, storageClusterName string) ([]*storagev1.StorageClass, error) {
+	var defaultStorageClasses []*storagev1.StorageClass
+
+	rbdName := fmt.Sprintf(defaultCephRBDStorageClassTemplate, storageClusterName)
```

Comment:
We also need the default virtualization storageclass.

Reply:
done